src/libs/actions/Policy/Policy.ts

```diff
@@ -3436,6 +3446,632 @@ function getAdminPoliciesConnectedToNetSuite(): Policy[] {
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.FORMS.RULES_AUTO_PAY_REPORTS_UNDER_MODAL_FORM,
+            value: {
+                maxExpenseAutoPayAmount: limit,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                autoReimbursement: {
+                    limit: parsedLimit,
+                },
+                pendingFields: {
+                    autoReimbursement: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.FORMS.RULES_AUTO_PAY_REPORTS_UNDER_MODAL_FORM,
+            value: {
+                maxExpenseAutoPayAmount: limit,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                autoReimbursement: {
+                    limit: parsedLimit,
+                },
+                pendingFields: {
+                    autoReimbursement: null,
+                },
+                errorFields: null,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                autoReimbursement: policy?.autoReimbursement,
+                pendingFields: {
+                    autoReimbursement: null,
+                },
+                errorFields: {
+                    autoReimbursement: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                },
+            },
+        },
+    ];
+
+    const parameters: SetPolicyAutoReimbursementLimitParams = {
+        autoReimbursement: {limit: parsedLimit},
+        policyID,
+    };
+
+    API.write(WRITE_COMMANDS.SET_POLICY_AUTO_REIMBURSEMENT_LIMIT, parameters, {
+        optimisticData,
+        successData,
+        failureData,
+    });
+}
+
+/**
+ * Call the API to enable auto-payment for the reports in the given policy
+ * @param policyID - id of the policy to apply the limit to
+ * @param enabled - whether auto-payment for the reports is enabled in the given policy
+ */
+function enablePolicyAutoReimbursementLimit(policyID: string, enabled: boolean) {
+    const policy = getPolicy(policyID);
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                shouldShowAutoReimbursementLimitOption: enabled,
+                autoReimbursement: {
+                    limit: enabled ? policy?.autoReimbursement?.limit : CONST.POLICY.AUTO_REIMBURSEMENT_DEFAULT_LIMIT_CENTS,
```

Comment:
Having a similar thought here about maybe leaving out the update if we are enabling